src/devtools/views/Components/TreeContext.js

```diff
@@ -381,13 +381,18 @@ function reduceSearchState(store: Store, state: State, action: Action): State {
             recursivelySearchTree(store, rootID, regExp, searchResults);
           });
 
-          if (searchResults.length > 0) {
+          const searchWithNearestResults = getNearestResults(
+            selectedElementID,
+            searchResults
+          );
+
+          if (searchWithNearestResults.length > 0) {
             if (prevSearchIndex === null) {
               searchIndex = 0;
```

Comment:
Seems like we probably don't need to create a whole new array above. I think this is the only case when we need to find the closest search match. I think we can just set a default `searchIndex` of something other than 0 if there's a `selectedElementID` and we find a "closer" match to it.

What do you think?

Reply:
I don't know if I understood what you meant, can you explain it differently?

Reply:
There's no need to create a second array. All we're doing is deciding what the best initial index is.